core/modules/syncer.js

```diff
@@ -509,119 +537,150 @@ Syncer.prototype.chooseNextTask = function() {
 		tiddlerInfo = this.tiddlerInfo[title];
 		tiddler = this.wiki.tiddlerExists(title) && this.wiki.getTiddler(title);
 		if(!tiddler) {
-			return new DeleteTiddlerTask(this,title);
+			deletePending.push(title);
 		}
 	}
 	// Finally, check for tiddlers that need loading
-	title = Object.keys(this.titlesToBeLoaded)[0];
-	if(title) {
-		delete this.titlesToBeLoaded[title];
-		return new LoadTiddlerTask(this,title);
-	}
+	loadPending.push(...Object.keys(this.titlesToBeLoaded));
 	// No tasks are ready now, but might be in the future
-	return havePending;
+	return {havePending: havePending, loadPending: loadPending, savePending: savePending, deletePending: deletePending, syncServer: syncServer};
 };
 
-function SaveTiddlerTask(syncer,title) {
+function SaveAllTiddlersTask(syncer,titles) {
 	this.syncer = syncer;
-	this.title = title;
+	this.titles = titles;
 	this.type = "save";
 }
 
-SaveTiddlerTask.prototype.toString = function() {
-	return "SAVE " + this.title;
-}
+SaveAllTiddlersTask.prototype.toString = function() {
+	return "SAVE ALL " + this.titles;
+};
 
-SaveTiddlerTask.prototype.run = function(callback) {
-	var self = this,
-		changeCount = this.syncer.wiki.getChangeCount(this.title),
-		tiddler = this.syncer.wiki.tiddlerExists(this.title) && this.syncer.wiki.getTiddler(this.title);
-	this.syncer.logger.log("Dispatching 'save' task:",this.title);
-	if(tiddler) {
-		this.syncer.syncadaptor.saveTiddler(tiddler,function(err,adaptorInfo,revision) {
-			// If there's an error, exit without changing any internal state
-			if(err) {
-				return callback(err);
-			}
-			// Adjust the info stored about this tiddler
-			self.syncer.tiddlerInfo[self.title] = {
-				changeCount: changeCount,
+SaveAllTiddlersTask.prototype.run = function(callback) {
+	this.syncer.logger.log("Dispatching 'save all' task:",this.titles);
+	var changeCounts = this.titles.reduce((r,title) => { r[title] = this.syncer.wiki.getChangeCount(title); return r; }, {});
+	var tiddlers = this.titles
+		.map(title => this.syncer.wiki.tiddlerExists(title) && this.syncer.wiki.getTiddler(title))
+		.filter(e => e);
```

Comment:
@Arlen22 ... I think we should have changed the ESLint rules instead of the code here. I personally think `(title) => ...` is much more readable than as it is now.